docs/overview/cmake-can-not.rst

```diff
@@ -13,21 +13,21 @@ What can't be done with CMake
 .. Can't find good reference. Let me know if can find a better link (Ruslo)
 
 :ref:`CMake <CMake>` has it's strengths and weaknesses. Most of the drawbacks
-mentioned here can be workarounded by using approaches that may differs from
-your normal workflow but reach the end goal. Try to look at them from another
-angle, think of the picture in a whole and remember that advantages definitely
+mentioned here can be worked around by using approaches that may differ from
+your normal workflow, yet still reach the end goal. Try to look at them from another
+angle; think of the picture as a whole and remember that the advantages definitely
 outweigh the disadvantages.
 
 Language/syntax
 ===============
 
-This is probably the first thing you will hit with. :ref:`CMake <CMake>`
+This is probably the first thing you will be hit with. The :ref:`CMake <CMake>`
 language is not something you can compare with what you have likely used
 before. There are no classes, no maps, no virtual functions or lambdas. Even
-parsing functions input arguments and returning results is something quite
+parsing the input arguments of each function then returning results is something quite
```

Comment:
> parsing the input arguments of each function **then** returning results

It's simply not accurate in terms of CMake. In C++ input arguments prepared before function called and returned values available after function's job finished, hence "then" is fine. By "parsing input arguments" in CMake I mainly mean usage of [cmake_parse_arguments](https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html) module and by "return result" I mean usage of [PARENT_SCOPE](https://cmake.org/cmake/help/latest/command/set.html#set-normal-variable). You can "return result" without/before/after "cmake_parse_arguments".

What about:
```
Even such tasks like "parse the input arguments of a function" and "return result from a function" are quite tricky for the beginners.
```
?

> parsing the input arguments of **each** function then returning results

There is no need to parse arguments for each function, can we use just "of a function" here?